### PR TITLE
Backport #2185 and #2196 to release-10.0

### PR DESCRIPTION
**Pull request by jrudolph**
And fix some Mima issues.

### CODE REVIEW

**Comment by raboof** [comment]
I can fix the paradox error

**Comment by jrudolph** [comment]
Thanks, there's probably a simple fix even if it's just using both `@scaladoc` and `@javadoc`?

**Comment by jrudolph** [comment]
Good to go from your side, @raboof?

File: akka-http-core/src/main/mima-filters/10.0.13.backwards.excludes [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# ToStrict is private[http]
+ProblemFilters.exclude[DirectMissingMethodProblem]("akka.http.impl.util.ToStrict.this")
+ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.http.javadsl.settings.ParserSettings.getMaxToStrictBytes")
+ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.http.javadsl.settings.ParserSettings.maxToStrictBytes")
+ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.http.scaladsl.settings.ParserSettings.maxToStrictBytes")
+ProblemFilters.exclude[DirectMissingMethodProblem]("akka.http.impl.settings.ParserSettingsImpl.*")
+ProblemFilters.exclude[IncompatibleResultTypeProblem]("akka.http.impl.settings.ParserSettingsImpl.*")
+
+# Uri conversion additions https://github.com/akka/akka-http/pull/1950
+ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.http.javadsl.model.Uri.asScala")
```
Comment (raboof) [inline akka-http-core/src/main/mima-filters/10.0.13.backwards.excludes]: :+1: moving these here

File: akka-http-core/src/main/scala/akka/http/impl/util/package.scala [hunk 2]
```diff
@@ -92,11 +92,12 @@ package object util {
...
 
-  import akka.http.scaladsl.model.{ ContentType, HttpEntity }
-  import akka.stream.{ Attributes, Outlet, Inlet, FlowShape }
+  import akka.http.scaladsl.model.{ ContentType, EntityStreamException, ErrorInfo, HttpEntity }
+  import akka.stream.{ Attributes, FlowShape, Inlet, Outlet }
+
   import scala.concurrent.duration.FiniteDuration
 
-  private[http] class ToStrict(timeout: FiniteDuration, contentType: ContentType)
+  private[http] class ToStrict(timeout: FiniteDuration, maxBytes: Option[Long], contentType: ContentType)
```
Comment (raboof) [inline akka-http-core/src/main/scala/akka/http/impl/util/package.scala]: Hum this can be a `Long` now, but private so we can do this later.

**Pull request merged**